electrons/python/generic_electron_solver.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import division
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from numpy.typing import NDArray
+from typing import Tuple
+import numpy as np
+
+
+@dataclass
+class GenericElectronSolver(ABC):
+    # Simulation parameters
+    electron_density_per_cm3: float  # fluence-rate [/cm^2/s]
+    voltage_V: float  # [V/cm] magnitude of the electric field
+    electrode_gap: float  # [cm] # electrode gap
+    unit_length_cm: float
```

Comment:
Maybe we should add a comment/rename the variable (e.g. grid_unit_lenght_cm, grid_resolution_cm, grid_step, grid_spacing_cm, etc.).